src/scripts/main.js

```diff
@@ -1,3 +1,345 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+const tableHeader = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+const form = document.createElement('form');
+
+const minAge = 18;
+const maxAge = 90;
+const minLength = 4;
+
+form.className = 'new-employee-form';
+
+form.insertAdjacentHTML('afterbegin',
+  `
+  <label>Name:
+    <input
+      name="name"
+      type="text"
+      data-qa="name"
+      required
+    >
+  </label>
+  <label>Position:
+    <input
+      name="position"
+      type="text"
+      data-qa="position"
+      required
+    >
+  </label>
+  <label>Office:
+    <select
+      name="office"
+      type="text"
+      data-qa="office"
+    >
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+  </label>
+  <label>Age:
+    <input
+      name="age"
+      type="number"
+      data-qa="age"
+      required
+    >
+  </label>
+  <label>Salary:
+    <input
+      name="salary"
+      type="number"
+      data-qa="salary"
+      required
+    >
+  </label>
+  <button type="submit">Save to table</button>
+  `
+);
+
+body.append(form);
+
+let check = false;
+
+tableHeader.addEventListener('click', e => {
+  const target = e.target;
+
+  sortTable(target.cellIndex, target.innerHTML, check === target.cellIndex);
+
+  check = (check === target.cellIndex) ? false : target.cellIndex;
+});
+
+function sortTable(colNum, attribute, directSorting) {
+  const rowsArray = [...tbody.children];
+
+  let sorter;
+
+  switch (attribute) {
+    case 'Position':
+    case 'Name':
+    case 'Office':
+      sorter = function(rowA, rowB) {
+        const stringA = rowA.cells[colNum].innerHTML;
+        const stringB = rowB.cells[colNum].innerHTML;
+
+        return stringA.localeCompare(stringB);
+      };
+      break;
+
+    default:
+      sorter = function(rowA, rowB) {
+        const num1 = toNumber(rowA.cells[colNum].innerHTML);
+        const num2 = toNumber(rowB.cells[colNum].innerHTML);
+
+        return num1 - num2;
+      };
+  }
+
+  rowsArray.sort(sorter);
+
```

Comment:
it looks useless for me)
```suggestion

  switch (attribute) {
    case 'Position':
    case 'Name':
    case 'Office':
      rowsArray.sort((rowA, rowB) => {
        const stringA = rowA.cells[colNum].innerHTML;
        const stringB = rowB.cells[colNum].innerHTML;

        return stringA.localeCompare(stringB);
      });
      break;

    default:
      rowsArray.sort((rowA, rowB) => {
        const num1 = toNumber(rowA.cells[colNum].innerHTML);
        const num2 = toNumber(rowB.cells[colNum].innerHTML);

        return num1 - num2;
      });
  }

```